### PR TITLE
[interop][SwiftToCxx] support std::string -> Swift::String conversion…

### PR DESCRIPTION
**Pull request by hyp**
… in C++

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Review by egorzhdan — APPROVED** [review]
Awesome!

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

File: lib/PrintAsClang/ModuleContentsWriter.cpp [hunk 1]
```diff
@@ -776,6 +776,10 @@ EmittedClangHeaderDependencyInfo swift::printModuleContentsAsCxx(
   writer.write();
   info.dependsOnStandardLibrary = writer.isStdlibRequired();
   if (M.isStdlibModule()) {
+    // Embed additional STL includes.
+    os << "#ifndef SWIFT_CXX_INTEROP_HIDE_STL_OVERLAY\n";
```
Comment (hyp) [inline lib/PrintAsClang/ModuleContentsWriter.cpp:780]: @Robertorosmaninho

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Pull request merged**